File: src/mvc/cpp/tree_model/main_window.cpp
#include "main_window.h"
#include "./ui_main_window.h"
#include "tree_model.h"
#include <QList>
#include <QStringList>

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent), ui(new Ui::MainWindow) {
  ui->setupUi(this);

  auto model = new TreeModel(QStringList() << "Col A" << "Col B" << "Col C");
  model->appendRow(QStringList() << "Elements");
  model->appendRow(QStringList() << "Items");
  ui->treeView->setModel(model);
}

MainWindow::~MainWindow() { delete ui; }

File: src/mvc/python/tree_model/main_window.py
from PyQt6 import QtWidgets, uic
from tree_model import TreeModel, TreeItem


class MainWindow(QtWidgets.QMainWindow):
    def __init__(self):
        super().__init__()
        uic.loadUi("main_window.ui", self)

        model = TreeModel()
        model.appendRow(TreeItem(["Elements"]))
        model.appendRow(TreeItem(["Items"]))
        self.treeView.setModel(model)

        self.show()
